fastmri/data/mri_data.py

```diff
@@ -358,3 +360,111 @@ def __getitem__(self, i: int):
             sample = self.transform(kspace, mask, target, attrs, fname.name, dataslice)
 
         return sample
+
+
+class AnnotatedSliceDataset(SliceDataset):
+    def __init__(
+            self,
+            root: Union[str, Path, os.PathLike],
+            challenge: str,
+            transform: Optional[Callable] = None,
+            use_dataset_cache: bool = False,
+            sample_rate: Optional[float] = None,
+            volume_sample_rate: Optional[float] = None,
+            dataset_cache_file: Union[str, Path, os.PathLike] = "dataset_cache.pkl",
+            num_cols: Optional[Tuple[int]] = None,
+            use_annotation: bool = False,
+            annotation_version: Optional[str] = None,
+            mri_type: Optional[str] = None,
+            multiple_annotation_policy: Optional[str] = None
+    ):
+        # subclass SliceDataset
+        SliceDataset.__init__(self, root, challenge, transform, use_dataset_cache, sample_rate, volume_sample_rate,
+                              dataset_cache_file, num_cols)
+
+        self.annotated_examples = []
+
+        if use_annotation is True:
+            # download csv file from github using git hash to find certain version
+            annotation_file_name = self.download_csv(annotation_version, mri_type)
+            annotations_csv = pd.read_csv(annotation_file_name)
+            for example in self.examples:
+                fname, slice_ind, metadata = example
+
+                # using filename and slice to find desired annotation
+                annotations_df = annotations_csv[
+                    (annotations_csv['file'] == fname.stem) & (annotations_csv['slice'] == slice_ind)]
+                annotations_list = annotations_df.itertuples(index=True, name='Pandas')
+
+                # if annotation (filename or slice) not found, fill in dummy values
+                if len(annotations_df) == 0:
+                    annotation = self.get_annotation(True, None)
+                    metadata['annotation'] = annotation
+                    self.annotated_examples += [(fname, slice_ind, metadata)]
+
+                else:
+                    # only use the first annotation
+                    if multiple_annotation_policy == 'first':
+                        rows = list(annotations_list)[0]
+                        annotation = self.get_annotation(False, rows)
+                        metadata['annotation'] = annotation
+                        self.annotated_examples += [(fname, slice_ind, metadata)]
+
+                    # use an annotation at random
+                    elif multiple_annotation_policy == 'random':
+                        random_number = random.randint(0, len(annotations_df) - 1)
```

Comment:
Don't use `random`. For parallel training, `random` might be duplicated across all processes so that all processes are picking the same random numbers. Instead, create a PyTorch generator object and sample the integer from that.

Reply:
Done. I used `torch.randint(len(df) - 1,(1,))` to sample the integer.